packages/patternfly-4/react-core/src/components/BackgroundImage/BackgroundImage.js

```diff
@@ -1,9 +1,33 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
 import { css } from '@patternfly/react-styles';
+import { css as emotionCSS, cx } from 'emotion';
 import styles from '@patternfly/patternfly-next/components/BackgroundImage/background-image.css';
 
+import backgroundImageFilter from '../../../styles/assets/images/background-filter.svg';
```

Comment:
These should not be part of react-core. This should be up to the consumer to give these paths. Having the imports here forces a consumer to setup an SVG loader and a file/URL loader for the jpg images. I do understand that becomes more work for the consumer, but it gives the freedom of hosting images in a CDNcdn, using webpack with URL loaders, copying to a public dir and using the URL, replacing and using their own, etc. Copying them to the styles dir is still a good idea so we can publish with the images, but we should not point to them here. A good component lib should not make assumptions on how files are loaded.

This should really just be an example in react-docs that is easy to copy/paste.  The example should make use of `patternfly/react-tokens` that way strings are not hardcoded as well.


If we really think it is too difficult for consumers to write a ~20 lines of code we can possibly script to output this or make a special import that does it. `import { setupImageUrls } from '@patternfly/react-core/webpack';` Then at least the consumer can opt into this behavior.

Last I talked with Ian from Insights there were plans to try and host images in a CDN so that they are cached better when switching from app to app. This would remove that possibility all together.

Reply:
We can require applications to defined their own image paths. However, users would also need to define this in order to use any component that makes use of backgroundImage; login layout, about modal, etc.

Also, it's not clear how we would use react-tokens in order to override the image paths? This is a string from react-tokens, which does not have a valid path for my example.

`export const c_background_image_BackgroundImage_xs = {"name":"--pf-c-background-image--BackgroundImage--xs","value":"url(assets/images/pfbg_576.jpg)","var":"var(--pf-c-background-image--BackgroundImage--xs)"} `

Reply:
I meant for the variable names
```js
const overrideCSS = `
${c_background_image_BackgroundImage_xs.name}: url('some url);
${c_background_image_BackgroundImage_sm.name}: url('some url);
`;
```

As for where they provide it they would only need to provide it once at a global level. or anywhere that has it's own path to an image. I guess I am still not a fan of background images or images in a component library since it over complicates things, but since we have to support it some way I would say we need to find a middle ground of having the consumer pass the values, but also making it pretty easy. To make things slightly easier we could strip out the default values for the variables when we run the babel plugin. This will make it easier to override because they are not declared anywhere. We can do this across the board for all pf css. Any css variable that has a value that starts with `url(` can be removed. They are all scoped to the compoennt anyways so there really is no reason for the default since it doesn't work anyways. We would still output them to react-tokens so they are easily referenced. That should remove the need to target the component in the css selector override. Users still can, but it will be optional.

If I was providing this component I would do have the following
```tsx
import styles from '@patternfly/patternfly-next/components/BackgroundImage/backgroundImage.css'; // has no variables for images declared. References remain

const ImageSizes = {
 xs: 'xs',
 sm: 'sm',
 sm2x: 'sm2x,
...etc
};

const variableMap = {
  [ImageSizes.xs]: c_background_image_BackgroundImage_xs,
  [ImageSizes.sm]: c_background_image_BackgroundImage_sm,
};

interface Props extends HTMLProps<HTMLDivElement> {
  src: string | { [K in ImageSizes]: ImageSizes[K] };
}

const BackgroundImage = ({ src, className, ... }) => {
 // default string value to xs to handle all sizes
  const variableOverrides = typeof src === 'string' ? { xs: src } : variableOverrides;
  cosnt  {  bgOverrides } = StyleSheet.create({
    bgOverrides: Object.keys(variableOverrides).reduce((prev, size) => {
         return {
            ...prev,
           [variableMap[size]: `url(${variableOverrides[size]})`
         }
    }, {})
  })
  return (
   <div className={css(styles.backgroundImage,bgOverrides, className)}  />
 )
}
```

Note that is psuedo code so it may not work copy/paste.

Consumers could then dor the override in 1 place and then rexport that component and consume it for the login layout, etc. 

Another option would be follow the same symantics as the `picture` html tag and so something like
```
<BackgroundImage>
  <BackgroundImageSource media="xs" src="pathToXS" />
  <BackgroundImageSource media="md" src="pathToMD" />
  <BackgroundImageSource media="lg" src="pathToMD" />
</BackgroundImage>
```
Then the `BackgroundImageSource` components take care of adding the overrides. AboutModal has a prop for the hero Image so that would reference the same paths. Login Layout should not have the background image as part of it. It is a layout and will not have background image component. LoginPage will, but that won't be provided by this lib. A demo will be made available that a user could swap out the backgroun image component with that with one that they setup.

Reply:
Ok, I'll try implementing your psuedo code. That looks like it will work well. Thank you.